script.js

```diff
@@ -1,4 +1,188 @@
+  var rolled =
+    `Welcome Player ${player}. <br>You rolled <u>${rollArray[0]} for Dice 1</u> and <u>${rollArray[1]} for Dice 2</u>. <br>Choose which dice to go first (1 or 2). ` +
+    leaderboard(scoreKeeper);
+  return rolled;
+}
+
+// ---------- LEADERBOARD ----------
+// checks which player is leading and list them in descending order
+function leaderboard(scoreKeeper) {
+  var scoreboard = "<br><br><u>Scoreboard</u><br>";
+  // total player count = scoreKeeper.length
+  if (scoreKeeper.length == 2) {
+    var display = `<br><br>Current score:<br>Player 1: ${scoreKeeper[0]}<br>Player 2: ${scoreKeeper[1]}`;
+    if (scoreKeeper[1] > scoreKeeper[0]) {
+      display = `<br><br>Current score:<br>Player 2: ${scoreKeeper[1]}<br>Player 1: ${scoreKeeper[0]}`;
+    }
+    return display;
+  } else {
+    for (p = 0; p < scoreKeeper.length; p++) {
+      var playerCounter = p + 1;
+      scoreboard =
+        scoreboard +
+        `Player ${playerCounter}: ${scoreKeeper[playerCounter - 1]}<br>`;
+    }
+    return scoreboard;
+  }
+}
+
+// ---------- SEQUENCING ----------
+// generates 2-digit number from input sequencing
+function sequencing(seq, rollArray, player) {
+  var sequence = "";
+  var finalNum = "";
+  console.log("sequencing fn " + seq);
+  console.log(rollArray, "in sequencing fn");
+  // user chooses which dice roll to be in front
+  if (isNaN(seq) || seq == 0) {
+    sequence =
+      `Please only enter 1 or 2<br><br>Player ${player}, you rolled <u>${rollArray[0]} for Dice 1</u> and <u>${rollArray[1]} for Dice 2</u>.` +
+      leaderboard(scoreKeeper);
+  }
+  if (seq <= 2 && seq > 0) {
+    if (seq == 1) {
+      finalNum = String(rollArray[0]) + String(rollArray[1]);
+      console.log(seq + " first: " + finalNum);
+    }
+    if (seq == 2) {
+      finalNum = String(rollArray[1]) + String(rollArray[0]);
+      console.log(seq + " first: " + finalNum);
+    }
+    // add 2-digit number as score to player
+    scoreKeeper[player - 1] += Number(finalNum);
+    sequence =
+      `Player ${player}, you chose Dice ${seq} first. <br>Your number is ${finalNum}. <br><br>It is now the next player's turn.` +
+      leaderboard(scoreKeeper);
+    // change mode
+    mode = "rolling";
+  }
+  return sequence;
+}
+
+// ========== MAIN FUNCTION ==========
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "Press submit to continue.";
+  var playerCount;
+
+  if (player <= scoreKeeper.length) {
+    console.log("current: " + player);
+    if (mode == "rolling") {
+      // reset array
+      playerRolls = [];
+      var store = storeNum(playerRolls, player);
+      return store;
+    }
+    if (mode == "sequencing") {
+      console.log(playerRolls);
+      var seqResult = sequencing(input, playerRolls, player);
+      // change player
+      player++;
+      return seqResult;
+    }
+  } else {
+    player = 1;
+    console.log("reset player count");
+  }
+
+  if (mode == "start") {
```

Comment:
If it's possible, try to put the start logic at the top of the main function so the flow of the game is more intuitive to a someone looking at the code for the 1st time.